src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncMainTests.cs

```diff
@@ -2063,5 +2063,397 @@ static async MyTask Main()
+            var comp = CreateEmptyCompilation(source, references: new[] { corlibRef }, options: TestOptions.DebugExe);
+            comp.MakeMemberMissing(SpecialMember.System_Runtime_CompilerServices_AsyncHelpers__HandleAsyncEntryPoint_Task);
+
+            var verifier = CompileAndVerify(comp, verify: Verification.Skipped);
+            // Verify it falls back to GetAwaiter().GetResult() pattern
+            verifier.VerifyIL("Program.<Main>()", """
+                {
+                  // Code size       19 (0x13)
+                  .maxstack  1
+                  .locals init (System.Runtime.CompilerServices.TaskAwaiter V_0)
+                  IL_0000:  call       "System.Threading.Tasks.Task Program.Main()"
+                  IL_0005:  callvirt   "System.Runtime.CompilerServices.TaskAwaiter System.Threading.Tasks.Task.GetAwaiter()"
+                  IL_000a:  stloc.0
+                  IL_000b:  ldloca.s   V_0
+                  IL_000d:  call       "void System.Runtime.CompilerServices.TaskAwaiter.GetResult()"
+                  IL_0012:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void AsyncMainFallbackToOldPattern_TaskOfInt()
+        {
+            var source = """
+                using System.Threading.Tasks;
+
+                class Program
+                {
+                    static async Task<int> Main()
+                    {
+                        await new Task<int>();
+                        return 42;
+                    }
+                }
+                """;
+
+            var corlibRef = CreateEmptyCompilation(MinimalAsyncCorelibWithAsyncHelpers, assemblyName: "mincorlib").EmitToImageReference();
+            var comp = CreateEmptyCompilation(source, references: new[] { corlibRef }, options: TestOptions.DebugExe);
+            comp.MakeMemberMissing(SpecialMember.System_Runtime_CompilerServices_AsyncHelpers__HandleAsyncEntryPoint_Task_Int32);
+
+            var verifier = CompileAndVerify(comp, verify: Verification.Skipped);
+            // Verify it falls back to GetAwaiter().GetResult() pattern
+            verifier.VerifyIL("Program.<Main>()", """
+                {
+                  // Code size       19 (0x13)
+                  .maxstack  1
+                  .locals init (System.Runtime.CompilerServices.TaskAwaiter<int> V_0)
+                  IL_0000:  call       "System.Threading.Tasks.Task<int> Program.Main()"
+                  IL_0005:  callvirt   "System.Runtime.CompilerServices.TaskAwaiter<int> System.Threading.Tasks.Task<int>.GetAwaiter()"
+                  IL_000a:  stloc.0
+                  IL_000b:  ldloca.s   V_0
+                  IL_000d:  call       "int System.Runtime.CompilerServices.TaskAwaiter<int>.GetResult()"
+                  IL_0012:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void AsyncMainFallbackToOldPattern_RedefinedTask()
```

Comment:
Why does this fall back? Isn't the Task used by Main the same as the Task used by HandleAsyncEntryPoint?